### DISCUSSION TITLE
How to transpose note along scale based on interval (across octaves)?

### DISCUSSION THREAD

**Discussion by braebo** [tonaljs/tonal#372] [Q&A]
Hey there!  First off- you all are amazing for making such an incredible library!!

So, I have a genereated array of notes in `C minor`, i.e. `['F4', 'G4', 'Bb4', 'Eb5', 'D5', 'C5']`

I want to loop over them and randomly transpose some of them through the scale:

```ts
const notes = `['F4', 'G4', 'Bb4', 'Eb5', 'D5', 'C5']`

for (const note of notes) {
  const interval = randomRange(-5, 5) // random int between -5 and 5
  
  const newNote = this.transposeNote(note, interval) // How would this function look?
}
```

I've tried a lot of different ways to get this working and I'm certain that I'm missing something obvious 😅.  For example, this abomination was one of many attempts but doesn't preserve the right octave and breaks if the interval is negative:
```ts
transposeNote(note: NoteObj, interval: number) {
    return [Scale.get(`C${note.octave} ${this.mode}`).notes.indexOf(`${note.name}${note.octave}`), interval].map(
        Scale.degrees(`C${note.ctave} ${this.mode}`)
    )[1]
}
```

The scale I'm using is always 1 of 4 options `C [minor | major | minor pentatonic | major pentatonic]`

My main problem is that `scale.notes` only returns 1 octave, but my notes span multiple octaves in the scale.

Is there a proper way to transpose notes along a scale based on a random interval?

**Comment by danigb** [comment]
```js
import {Scale, Range}  from 'tonal';

const scale = Scale.degrees('c4 major');

Range.numeric([1, 20]).map(scale); // => 
// => ["C4", "D4", "E4", "F4", "G4", "A4", "B4", "C5", "D5", "E5", "F5", "G5", "A5", "B5", "C6", "D6", "E6", "F6", "G6", "A6"]
```

See https://jsfiddle.net/6sng9dbt/
